components/PortalStatistics.lua

```diff
@@ -0,0 +1,1407 @@
+
+
+---@param args table
+---@return Html
+function StatisticsPortal.topEarningsChart(args)
+	args = args or {}
+	args.limit = tonumber(args.limit) or 10
+
+	local params = {
+		catLabel = Logic.readBool(args.isForTeam) and 'Teams' or 'Players',
+		flipAxes = true,
+		emphasis = 'none',
+	}
+
+	local config = StatisticsPortal._getChartConfig(args, params)
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{ConditionNode(ColumnName('earnings'), Comparator.gt, 0)}
+	local topEarningsList
+
+	if config.opponentType == Opponent.team then
+		topEarningsList = StatisticsPortal._getTeams(
+			args.limit,
+			conditions:toString(),
+			'sum::earnings desc',
+			'earnings asc')
+	elseif config.opponentType == Opponent.solo then
+		topEarningsList = StatisticsPortal._getPlayers(
+			args.limit,
+			conditions:toString(),
+			'sum::earnings desc',
+			'earnings asc')
+	end
+
+	local yearSeriesData = Array.map(Array.range(config.startYear, CURRENT_YEAR), function(year)
+		return Array.map(Array.reverse(topEarningsList), function(teamData)
+			return teamData.extradata['earningsin' .. year] or 0
+		end)
+	end)
+
+	local opponentNames = Array.map(Array.reverse(topEarningsList), function(opponent)
+		return config.opponentType == Opponent.team and opponent.name or opponent.id
+	end)
+
+	if Logic.readBool(config.yearBreakdown) then
+		return StatisticsPortal._buildChartData(config, yearSeriesData, opponentNames)
+	else
+		local chartData = {}
+		chartData[1] = {
+			name = 'Total Earnings',
+			type = 'bar',
+			stack = config.stackType,
+			data = StatisticsPortal._addArrays(yearSeriesData),
+		}
+
+		config.yAxis = {
+			type = 'value',
+			name = 'Earnings ($USD)',
+		}
+		config.xAxis = {
+			type = 'category',
+			name = config.catLabel,
+			data = opponentNames,
+			axisTick = {
+				alignWithLabel = true,
+			},
+		}
+		config.customLegend = config.customLegend or config.customInputs
+		return StatisticsPortal._drawChart(config, chartData)
+	end
+end
+
+
+--[[
+Section: Coverage Breakdown
+]] --
+
+
+---@param args table
+---@return Html
+function StatisticsPortal.coverageStatistics(args)
+	args = args or {}
+	args.alignSide = Logic.readBool(args.alignSide)
+
+	local wrapper = mw.html.create('div')
+	local tournamentTable = wrapper:tag('div')
+	local matchTable = wrapper:tag('div')
+
+	if args.alignSide then
+		tournamentTable
+			:addClass('template-box')
+			:css('padding-right', '2em')
+		matchTable
+			:addClass('template-box')
+			:css('padding-right', '2em')
+	end
+
+	tournamentTable:node(StatisticsPortal.coverageTournamentTable(args))
+	matchTable:node(StatisticsPortal.coverageMatchTable(args))
+
+	return wrapper
+end
+
+
+---@param args table
+---@return Html
+function StatisticsPortal.coverageMatchTable(args)
+	args = args or {}
+	args.multiGame = Logic.readBool(args.multiGame)
+	args.customGames = (type(args.customGames) == 'table' and args.customGames)
+		or StatisticsPortal._splitOrDefault(args.customGames --[[@as string]], GAMES)
```

Comment:
```suggestion
		or StatisticsPortal._splitOrDefault(args.customGames, GAMES)
```
nil is allowed ...

Reply:
could you screenshot the warning in vscode, i am on mobile for the rest of today

Reply:
I'm afk til tomorrow but I'll screenshot it tomorrow